jxl/src/frame/modular.rs

```diff
@@ -86,15 +86,50 @@ impl ModularGridKind {
     }
 }
 
+// All the information on a specific buffer needed by Modular decoding.
+#[derive(Debug)]
+struct ModularChannel {
+    // Actual pixel buffer.
+    data: Image<i32>,
+    // Holds additional information such as the weighted predictor's error channel's last row for
+    // the transform chunk that produced this buffer.
+    auxiliary_data: Option<Image<i32>>,
+    // Shift of the channel (None if this is a meta-channel).
+    shift: Option<(usize, usize)>,
+}
+
+impl ModularChannel {
+    fn new(size: (usize, usize)) -> Result<Self> {
+        Self::new_with_shift(size, Some((0, 0)))
+    }
+
+    fn new_with_shift(size: (usize, usize), shift: Option<(usize, usize)>) -> Result<Self> {
```

Comment:
Is this ever going to be called with `shift = None`? Would it make sense to take a naked tuple directly?

Reply:
Not right now, but I expect it might be in the (near) future